### PR TITLE
Update allure-codeception in order to support php8

### PR DESCRIPTION
**Pull request by ihor-sviziev**
### Description

This PR updates `allure-codeception` to version 1.5 in order to bring compatibility with PHP8. It is needed in order to resolve dependencies issue for https://github.com/magento/magento2/pull/32614

### Fixed Issues (if relevant)

1. magento/magento2-functional-testing-framework#<issue_number>: Issue title
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
         "ext-intl": "*",
         "ext-json": "*",
         "ext-openssl": "*",
-        "allure-framework/allure-codeception": "~1.4.0",
+        "allure-framework/allure-codeception": "~1.5.0",
```
Comment (ihor-sviziev) [inline composer.json:18]: Also for backward compatibility we might keep something like this:
```suggestion
        "allure-framework/allure-codeception": "~1.4.0||~1.5.0",
```

**Comment by ihor-sviziev** [comment]
Looks like unit test failures not related to changes from this PR.

**Comment by jilu1** [comment]
@magento import pull request to https://github.com/magento-commerce/magento2-functional-testing-framework

**Comment by jilu1** [comment]
@ihor-sviziev since our build infrastructure is not currently support php8 and other php8 work is still WIP, we have created a new branch **php8-develop** and this PR is merged into that branch. Please let us know if this works for you.

**Comment by ihor-sviziev** [comment]
@jilu1 but somehow it's still not showing merged. Should I change target to that branch ?

**Comment by jilu1** [comment]
@ihor-sviziev I think you are right. I just reverted your commit from **php8-develop**, after it's sync with public repo (you might want to wait a bit), you can change PR to base from **php8-develop**.

**Comment by jilu1** [comment]
@ihor-sviziev you can fix the base of this PR now.

**Comment by ihor-sviziev** [comment]
@jilu1 changed target to php8-develop

**Comment by jilu1** [comment]
@magento import pull request to https://github.com/magento-commerce/magento2-functional-testing-framework

**Comment by magento-engcom-team** [comment]
@jilu1 the pull request successfully imported.

**Pull request merged**